src/struct.jl

```diff
@@ -50,6 +50,26 @@ axiskeys(x::KeyedVector) = tuple(getindex(getfield(x, :keys)))
 axiskeys(x::KeyedArray, d::Int) = d<=ndims(x) ? getindex(axiskeys(x), d) : OneTo(1)
 axiskeys(x::KeyedVector, d::Int) = d==1 ? getindex(getfield(x, :keys)) : OneTo(1)
 
+"""
+    named_axiskeys(arr)::NamedTuple
+
+Return the [`axiskeys`](@ref) along with their names.
+If there are duplicate names or unnamed axes, an error is thrown.
+
+```jldoctest
+julia> using AxisKeys
+
+julia> arr = KeyedArray([1 2], x=[1], y=[2,3]);
+
+julia> named_axiskeys(arr)
+(x = [1], y = [2, 3])
+```
+"""
+function named_axiskeys(arr)::NamedTuple
+    NT = NamedTuple{dimnames(arr)}
+    return NT(axiskeys(arr))
```

Comment:
```suggestion
function named_axiskeys(x)
    NT = NamedTuple{dimnames(x)}
    NT(keys_or_axes(x))
```
Not 100% sure, but with `keys_or_axes` it still returns something on a NamedDimsArray.

Reply:
Since `axiskeys` does not work on `NamedDimsArray`, it does seem consistent that `named_axiskeys` does not work either?

Reply:
Yes that may be more logical.

Reply:
Ok I will then leave it as is.